### ISSUE TITLE
estimate_steps_per_epoch ignores quadratic_duration, so a derived epoch covers ~54% of the data and the logged epoch count is wrong

### ISSUE THREAD

**Issue by g8a9** [MELT-proj/training#96]
## What happens

`estimate_steps_per_epoch` (`melt/training/data/audio/lhotse/dataloader.py`) derives the number of batches in an epoch as:

```python
batches_per_epoch = math.ceil(total_duration / batch_duration)
```

i.e. it assumes every batch is filled to `batch_duration` seconds of **real audio**. It is not. When `quadratic_duration` is set — every ablation-campaign config sets it to 30 — lhotse's `TimeConstraint` charges each cut

```
cost(d) = d + d² / quadratic_duration
```

so a batch reaches its budget well short of `batch_duration` seconds of actual audio. A 30 s cut costs 60; a 10 s cut costs 13.3. The estimator models none of this, so it **underestimates the batches in an epoch** and therefore the steps.

## Measured

MN5 job `44947472`, `ABL-MA-125-asr.yaml` at world_size 8, `batch_duration: 120`, `gradient_accumulation_steps: 4`, `quadratic_duration: 30`.

Nominal throughput implied by the estimator: `120 s × 8 ranks × 4 accum = 3840 s = 1.067 h` per optimizer step.

Actual, from the trainer's own `train_hours/total` counter over 55 consecutive steps (eleven logging intervals, low variance, well past warmup):

**0.571 h per step — 53.6% of nominal.**

| | value |
|---|---|
| estimator: steps/epoch at ws=8 | 586 |
| audio those 586 steps actually cover | ~335 h |
| the mix's real size | 625 h |
| fraction of a true pass | **~0.54** |
| steps for a true single pass | ~1094 |

## Why it matters

1. **`num_train_epochs` does not mean what it says.** With `num_train_epochs: 1` the trainer runs 586 steps and logs `epoch: 1.0` having seen roughly 54% of the data. Anyone reading the logs, W&B, or `trainer_state.json` will believe a full pass happened.
2. **It silently rescales the LR schedule.** `max_steps` derived from epochs sets the cosine horizon, so warmup and decay are stretched over a step count that does not correspond to the intended data exposure — the same failure mode `infra/check_training_config.py` warns about for a stale `total_hours`.
3. **The error is config-dependent.** It scales with `quadratic_duration` and with the duration distribution of the mix, so two configs are skewed by *different* factors. Arms sharing one config stay comparable; arms with different `quadratic_duration` or different corpus mixes do not.

The docstring's stated formula is exactly the arithmetic being questioned, so the function is doing what it says — the gap is that what it says has not been true since `quadratic_duration` entered the configs.

## Suggested fix

The honest fix is to stop predicting the fill and measure it, but that costs a pass over the data. Cheaper options:

1. **Apply the same cost function the sampler uses.** Estimate `E[cost(d)] / E[d]` from the cut durations already being read for `total_hours`, and divide `batches_per_epoch` by that ratio. Needs a duration histogram rather than just the total, but `compute_dataset_duration` is already walking the manifests.
2. **Calibrate at runtime.** After N steps, compare `train_hours/total` against the prediction and log a loud warning when they diverge by more than, say, 10%. Does not fix the LR horizon, but it turns a silent 2x into a visible one, and it is a handful of lines.
3. **Add a check to `infra/check_training_config.py`** flagging any config that sets both `batch_duration` and `quadratic_duration`, noting that the derived step count will be optimistic.

(2) and (3) are worth doing regardless of whether (1) lands — right now nothing anywhere reports the discrepancy.

## Current campaign status

Campaign arms now run `num_train_epochs: 1` with derived steps (#93). The bias is left uncompensated on purpose, because it is identical for every arm and so preserves step-for-step comparability. That decision only holds while every arm shares `quadratic_duration` and the same mix, which is worth remembering if a later phase varies either.
